tls/s2n_connection.h

```diff
@@ -148,6 +148,9 @@ struct s2n_connection {
      */
```

Comment:
I think a comment here on how mfl affects max_outgoing_fragment_length would be useful